packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,2841 @@
+
+  /// Creates a const [MenuAnchor] that can be positioned instead of aligned to
+  /// the region.
+  ///
+  /// Using this constructor also turns off the [TapRegion] mechanism present
+  /// when the other constructors are used, allowing menus created with this
+  /// constructor to close automatically when the [MenuAnchor] region (or any
+  /// other non-menu region) is clicked.
+  ///
+  /// The [style] defaults to a [MenuStyle.alignment] of [Alignment.topLeft],
+  /// since that is what is necessary for the positioning of the menu to be in
+  /// the correct coordinate space. The `position` argument to
+  /// [MenuController.open] is used to position the menu in the coordinate space
+  /// of the [MenuAnchor] when it is opened. It defaults to [Offset.zero] if no
+  /// `position` is given.
+  ///
+  /// The [menuChildren] argument is required.
+  const MenuAnchor.positioned({
+    super.key,
+    this.controller,
+    this.focusNode,
+    this.style = const MenuStyle(alignment: Alignment.topLeft),
+    this.onOpen,
+    this.onClose,
+    this.clipBehavior = Clip.none,
+    required this.menuChildren,
+    this.builder,
+    this.child,
+  })  : _isBar = false,
+        _isPositioned = true,
+        alignmentOffset = Offset.zero;
+
+  /// Creates a [MenuAnchor] that places its [menuChildren] into a [Flex]
+  /// container instead of popping them up in an overlay like the default
+  /// constructor does.
+  ///
+  /// Menus created in this way can't be "opened" or "closed", since it's part
+  /// of the regular widget tree, and it doesn't have a [builder] or a [child],
+  /// since those will be determined by the contents of [menuChildren].
+  ///
+  /// This is also used by [MenuBar] to create a menu bar.
+  ///
+  /// The [menuChildren] argument is required.
+  ///
+  /// See also:
+  ///
+  ///  * [MenuBar], a widget that creates a menu bar with cascading submenus.
+  const MenuAnchor.bar({
+    super.key,
+    this.controller,
+    this.style,
+    this.alignmentOffset = Offset.zero,
+    this.clipBehavior = Clip.none,
+    required this.menuChildren,
+  })  : onOpen = null,
+        builder = null,
+        child = null,
+        onClose = null,
+        focusNode = null,
+        _isBar = true,
+        _isPositioned = false;
+
+  final bool _isBar;
+  final bool _isPositioned;
+
+  /// An optional controller that allows opening and closing of the menu from
+  /// other widgets.
+  final MenuController? controller;
+
+  /// The [focusNode] attribute is the optional [FocusNode] also associated the
+  /// [child] or [builder] widget that opens the menu.
+  ///
+  /// The focus node should be attached to the widget that should receive focus
+  /// if keyboard focus traversal moves the focus off of the submenu with the
+  /// arrow keys.
+  ///
+  /// If not supplied, then keyboard traversal from the menu back to the
+  /// controlling button when the menu is open is disabled.
+  final FocusNode? focusNode;
```

Comment:
I wonder if this should be renamed. When seeing a `focusNode` property I expect that it's the focus node for that thing. But here, it is the focus node for another thing. That gets a little confusing (see also my comment in the sample code).